lib/Sema/CSFix.h

```diff
@@ -862,6 +866,25 @@ class AllowInvalidInitRef final : public ConstraintFix {
                                      ConstraintLocator *locator);
 };
 
+class AllowTupleTypeMismatch final : public ConstraintFix {
```

Comment:
I think it should inherit from `ContextualMismatch` that way you can remove `LHS, RHS` since parent would already have storage for them. One more initializer could be added to `ContextualMismatch` to support a custom `FixKind` passed in as a parameter.